### PR TITLE
docs(cli): add docs on how the CLI detects artifacts

### LINKED ISSUE #1085: Document how is the CLI generator discovering models and repositories in the project
See https://github.com/strongloop/loopback-next/issues/727#issuecomment-359700960:

> I just realised that our documentation added by strongloop/loopback.io#583 is missing an important piece of information: how is the generator discovering models and repositories in the project? What conventions does it require to work correctly?

## Acceptance criteria

 - [ ] In [REST Controller with CRUD Methods](http://loopback.io/doc/en/lb4/Controller-generator.html) ([source](https://github.com/strongloop/loopback-next/blob/3db07eb556c5462b5ac169dd2eae9b5db59bfaf4/docs/site/Controller-generator.md#rest-controller-with-crud-methods)), add a new subsection describing how the CLI is discovering models and repositories. What files are scanned, how are model/repository names inferred. Are we parsing the typescript to find all exports, as `@loopback/boot` does? Are we building the class names from file names? Are there any filters (e.g. does a repository class name have to include `Repository` suffix)? Etc.
 - [ ] In CLI, after the user selects "REST Controller with CRUD Methods" type, but before prompting for model & repository, print a short message with a link to this new documentation. Alternatively, instead of printing this hint to all users, print it only when no models or no repositories were found. The goal is to provide a clear next step for users that run into the problem that (one of|none of) their (models|repositories) was discovered by our CLI.

### PR DESCRIPTION
**Pull request by shimks**
Divided into two commits:
- the first includes the link to the documentation on CLI to the error thrown when no models/repositories are detected AND switches promises to use es7's async/await
- the second is the actual docs

Closes #1085

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: docs/site/Controller-generator.md [hunk 1]
```diff
@@ -69,6 +69,20 @@ to select:
 - The repository for this model that provides datasource connectivity
 - The REST path name to host the endpoints on
 
+The prompts that list out the models and repositories to choose from to build
+the controller with are chosen from the existing model/repository files on disc.
+From the LoopBack 4 project that the CLI is run, the CLI tool will search for
+file names ending with the following strings under `src/models` and
+`src/repositories`: `model.ts`/`model.js` and `repository.ts`/`repository.ts`.
```
Comment (bajtos) [inline docs/site/Controller-generator.md]: I find it difficult to understand what does this sentence mean. How about using a wildcard approach instead?

```markdown
The CLI tool will search for the following files in the LoopBack 4 project it runs in:
 - `src/repositories/*.repository.ts` and 
 - `src/repositores/*.repository.js` to find model repositories
 - `src/models/*.model.ts` and 
 - `src/models/*.model.js` to find model classes
```

File: docs/site/Controller-generator.md [hunk 2]
```diff
@@ -69,6 +69,20 @@ to select:
...
+Files that match these patterns will then be identified based on the string
+before the first `.` separator.
```
Comment (bajtos) [inline docs/site/Controller-generator.md]: Please add an example.

```markdown
before the first `.` separator. For example, file `models/product.model.ts` is identified 
as a source of `Product` model.
```

File: packages/cli/generators/controller/index.js [hunk 3]
```diff
@@ -88,92 +88,95 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
...
+    const repositoryList = await utils.getArtifactList(
+      this.artifactInfo.repositoryDir,
+      'repository',
+      true,
+    );
+    if (_.isEmpty(modelList)) {
+      throw new Error(
+        `No models found in ${
+          this.artifactInfo.modelDir
+        }. Please visit htpp://loopback.io/doc/en/lb4/Controller-generator.html for information on how models are discovered`,
```
Comment (bajtos) [inline packages/cli/generators/controller/index.js]: This line will be very long when printed to terminal. Please consider adding one or two line breaks, e.g. after the first sentence and then after the URL.

```
No models found in models.
Please visit htpp://loopback.io/doc/en/lb4/Controller-generator.html
for information on how models are discovered
```

File: packages/cli/generators/controller/index.js [hunk 4]
```diff
@@ -88,92 +88,95 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
...
+      );
+    }
+    if (_.isEmpty(repositoryList)) {
+      throw new Error(
+        `No repositories found in ${
+          this.artifactInfo.repositoryDir
+        }. Please visit htpp://loopback.io/doc/en/lb4/Controller-generator.html for information on how repositories are discovered`,
```
Comment (bajtos) [inline packages/cli/generators/controller/index.js]: Ditto, please consider breaking this long text into multiple lines.

File: docs/site/Controller-generator.md [hunk 5]
```diff
@@ -69,6 +69,26 @@ to select:
 - The repository for this model that provides datasource connectivity
 - The REST path name to host the endpoints on
 
+The prompts that list out the models and repositories to choose from to build
+the controller with are chosen from the existing model/repository files on disc.
+From the LoopBack 4 project that the CLI is run, The CLI tool will search for
```
Comment (virkt25) [inline docs/site/Controller-generator.md]: CLI is run, The CLI tool will search => CLI is run **in, the** CLI ...

File: packages/cli/generators/controller/index.js [hunk 6]
```diff
@@ -88,92 +88,97 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
       });
   }
 
-  promptArtifactCrudVars() {
-    let modelList = [];
-    let repositoryList = [];
+  async promptArtifactCrudVars() {
+    // let modelList = [];
```
Comment (virkt25) [inline packages/cli/generators/controller/index.js]: You can delete this line and the one below.

**Pull request merged**